python/cuml/dask/datasets/regression.py

```diff
@@ -40,9 +45,173 @@ def create_rs_generator(random_state):
     return rs
 
 
-def make_low_rank_matrix(n_samples=100, n_features=100, effective_rank=10,
-                         tail_strength=0.5, random_state=None, n_parts=1,
-                         n_samples_per_part=None):
+def _f_order_standard_normal(nrows, ncols, dtype, seed):
+    local_rs = cp.random.RandomState(seed=seed)
+    x = local_rs.standard_normal(nrows * ncols, dtype=dtype)
+    x = x.reshape((nrows, ncols), order='F')
+    return x
+
+
+def f_order_standard_normal(client, rs, chunksizes, ncols, dtype):
+    workers = list(client.has_what().keys())
+
+    n_chunks = len(chunksizes)
+    chunks_workers = (workers * n_chunks)[:n_chunks]
+
+    chunk_seeds = rs.permutation(len(chunksizes))
+    chunks = [client.submit(_f_order_standard_normal, chunksize, ncols, dtype,
+                            chunk_seeds[idx], workers=[chunks_workers[idx]],
+                            pure=False)
+              for idx, chunksize in enumerate(chunksizes)]
+
+    chunks_dela = [da.from_delayed(dask.delayed(chunk),
+                                   shape=(chunksizes[idx], ncols),
+                                   meta=cp.zeros((1)), dtype=dtype)
+                   for idx, chunk in enumerate(chunks)]
+    return da.concatenate(chunks_dela, axis=0)
+
+
+def _data_from_multivariate_normal(seed, covar, n_samples, n_features, dtype):
+    mean = cp.zeros(n_features)
+    local_rs = cp.random.RandomState()
+    return local_rs.multivariate_normal(mean, covar, n_samples,
+                                        dtype=dtype)
+
+
+def data_from_multivariate_normal(client, rs, covar, chunksizes, n_features,
+                                  dtype):
+    workers = list(client.has_what().keys())
+
+    n_chunks = len(chunksizes)
+    chunks_workers = (workers * n_chunks)[:n_chunks]
+
+    chunk_seeds = rs.permutation(len(chunksizes))
+
+    data_parts = [client.submit(_data_from_multivariate_normal,
+                                chunk_seeds[idx], covar,
+                                chunksizes[idx], n_features,
+                                dtype, workers=[chunks_workers[idx]],
+                                pure=False)
+                  for idx, chunk in enumerate(chunksizes)]
+
+    data_dela = [da.from_delayed(dask.delayed(chunk),
+                                 shape=(chunksizes[idx], n_features),
+                                 meta=cp.zeros((1)), dtype=dtype)
+                 for idx, chunk in enumerate(data_parts)]
+    return da.concatenate(data_dela, axis=0)
+
+
+def _f_order_shuffle(X, y, n_samples, seed, features_indices):
+    local_rs = cp.random.RandomState(seed=seed)
+    samples_indices = local_rs.permutation(n_samples)
+
+    X[...] = X[samples_indices, :]
+    X[...] = X[:, features_indices]
+
+    y[...] = y[samples_indices, :]
+    return X, y
+
+
+def f_order_shuffle(client, rs, X, y, chunksizes, n_features,
+                    features_indices, n_targets, dtype):
+    X_parts = client.sync(_extract_partitions, X)
```

Comment:
It would be nice to start using the distributed data handler instead of calling `client.sync` directly so that we can guarantee that code path. This will also simplify your code so that the corresponding X and y parts can be passed directly to the `_f_order_shuffle`.